controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java

```diff
@@ -131,23 +128,31 @@ public void createScope(final CreateScopeRequest createScopeRequest, final Secur
         .thenAccept(x -> LoggerHelpers.traceLeave(log, "createScope", traceId));
     }
 
-    private void authenticate(String resourceName, AuthHandler.Permissions level) throws AuthenticationException {
+    private boolean authenticate(String resourceName, AuthHandler.Permissions level, AsyncResponse asyncResponse) {
         if (pravegaAuthManager != null ) {
             Map<String, String> map = null;
             List<String> authParams = headers.getRequestHeader(HttpHeaders.AUTHORIZATION);
 
             if (authParams == null || authParams.isEmpty()) {
-                throw new AuthenticationException("Auth failed for " + resourceName);
+                asyncResponse.resume(Response.status(Status.UNAUTHORIZED).build());
+                return false;
             }
 
             String authHeader = authParams.get(0);
             map = Arrays.stream(authHeader.split(",")).map(str -> str.split(":"))
                                             .collect(Collectors.toMap(e -> e[0], e -> e[1]));
 
-            if (!pravegaAuthManager.authenticate(resourceName, map, level)) {
-                throw new AuthenticationException("Auth failed for " + resourceName);
+            try {
+                if (!pravegaAuthManager.authenticate(resourceName, map, level)) {
```

Comment:
401 means unauthenticated but for some reason the constants in http have always said "unauthorized".  it's confusing but I guess that's how it is.  and 403 is forbidden (which should really be called unauthorized but again that's how it is:))

Reply:
I like much better the use of exceptions in this case because it less prone to errors. If by accident the developer forgets to check the boolean return value,  then the exception will break the regular flow. I suggest that we leave the exception and add one for authorization.

Reply:
Fixed. Added another exception and the method does not return a boolean now.